### ISSUE TITLE
Create a focus-trap functionality

### ISSUE THREAD

**Issue by kacperknapik** [SAP/cloud-commerce-spartacus-storefront#5945]
To be applied in:

- modals
- hamburger menu
- ... ?

**Comment by kacperknapik** [comment]
Modals currently have focus trap from bootstrap. This ticket can be finished after we remove bootstrap.

**Comment by Zeyber** [comment]
QA Steps:
1 - Add `FocusTrapModule` to imports in `product-list.module.ts`
2 - Add `cx-focus-trap` directive to `<div class="cx-search-facet">` in `product-facet-navigation.component.html`
3 - Navigate to any product list page with facets on it.
4 - Press tab key until focus is in `<div class="cx-search-facet">`
5 - Check you cannot focus elements outside of this element using only `tab` or `shift+tab` keys

**Comment by kacperknapik** [comment]
**QA results**
- used dev12
- improved autofocus

Works as expected
